test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_core_test.clj

```diff
@@ -301,3 +311,53 @@
           "Unexpected port for second connector")
       (is (= 42 (.getSoLingerTime (second connectors)))
           "Unexpected so linger time for second connector"))))
+
+(deftest test-idle-timeout
+  (let [read-lines (fn [r]
+                     (let [sb (StringBuilder.)]
+                       (loop [l (.readLine r)]
+                         (when l
+                           (.append sb l)
+                           (.append sb "\n")
+                           ;; readLine will block until the socket is closed,
+                           ;; or will throw a SocketTimeoutException if there
+                           ;; is no data available within the SoTimeout value.
+                           (recur (.readLine r))))
+                       (.toString sb)))
+        body "Hi World\n"
+        path "/hi_world"
+        ring-handler (fn [req] {:status 200 :body body})
+        read-response (fn [client-so-timeout]
+                        (let [s (Socket. "localhost" 9000)
+                              out (PrintWriter. (.getOutputStream s) true)]
+                          (.setSoTimeout s client-so-timeout)
+                          (.println out (str "GET " path " HTTP/1.1\n"
+                                             "Host: localhost\n"
+                                             "\n"))
+                          (let [in (BufferedReader. (InputStreamReader. (.getInputStream s)))]
+                            (read-lines in))))]
+    (let [config {:webserver {:port 9000
+                              :host "localhost"
+                              :idle-timeout-milliseconds 500}}]
+      (with-test-logging
+        (with-app-with-config app
+          [jetty9-service]
+          config
+          (let [s (tk-app/get-service app :WebserverService)
+                add-ring-handler (partial add-ring-handler s)]
+            (add-ring-handler ring-handler path)
+
+            (testing "Verify that server doesn't close socket before idle timeout"
+              ;; if we set the client socket timeout lower than the server
+              ;; socket timeout, we should get a timeout exception from the
+              ;; client side while attempting to read from the socket.
+              (is (thrown-with-msg? SocketTimeoutException #"Read timed out"
```

Comment:
I've run this test 50 or so times locally without seeing it fail.  It does seem like it could fail given that the server might start its idle timeout counter a bit sooner after writing the last byte of the response to the client than the client would after consuming the last byte of the response received back from the server.  I'm okay leaving this in for now, but we may need to revisit this if the test starts failing in CI.